ppl/zqd/apiserver/db.go

```diff
@@ -0,0 +1,40 @@
+package apiserver
+
+import (
+	"context"
+
+	"github.com/brimsec/zq/api"
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/ppl/zqd/postgres"
+)
+
+type DB interface {
+	CreateSpace(context.Context, SpaceRow) error
+	CreateSubspace(context.Context, SpaceRow) error
+	GetSpace(context.Context, api.SpaceID) (SpaceRow, error)
+	ListSpaces(context.Context) ([]SpaceRow, error)
+	UpdateSpaceName(context.Context, api.SpaceID, string) error
+	DeleteSpace(context.Context, api.SpaceID) error
+}
+
+type DBKind string
+
+const (
+	DBUnspecified DBKind = ""
+	DBFile        DBKind = "file"
+	DBPostgres    DBKind = "postgres"
+)
+
+type DBConfig struct {
+	Kind     DBKind          `json:"kind"`
+	Postgres postgres.Config `json:"postgres"`
```

Comment:
Is this ever marshaled? If not, no need for the struct tags.

Reply:
I suppose not. Will remove.